stdlib/public/core/Span/Span.swift

```diff
@@ -12,236 +12,270 @@
+extension Span where Element: ~Copyable {
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
-  /// The memory in `buffer` must be owned by the instance `owner`,
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
   ///
   /// - Parameters:
   ///   - buffer: an `UnsafeBufferPointer` to initialized elements.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
   public init(
     _unsafeElements buffer: UnsafeBufferPointer<Element>
-  ) -> dependsOn(immortal) Self {
-    precondition(
-      buffer.count == 0 || buffer.baseAddress.unsafelyUnwrapped.isAligned,
-      "baseAddress must be properly aligned for accessing \(Element.self)"
+  ) {
+    _precondition(
+      ((Int(bitPattern: buffer.baseAddress) &
+        (MemoryLayout<Element>.alignment&-1)) == 0),
+      "baseAddress must be properly aligned to access Element"
     )
-    self.init(_unchecked: buffer)
+    self.init(_unchecked: buffer.baseAddress, count: buffer.count)
   }
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
-  /// The memory in `buffer` must be owned by the instance `owner`,
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// The memory in `buffer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
   ///
   /// - Parameters:
   ///   - buffer: an `UnsafeMutableBufferPointer` to initialized elements.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
   public init(
     _unsafeElements buffer: UnsafeMutableBufferPointer<Element>
-  ) -> dependsOn(immortal) Self {
+  ) {
     self.init(_unsafeElements: UnsafeBufferPointer(buffer))
   }
 
-  /// Unsafely create a `Span` over initialized memory.
+  /// Unsafely creates a `Span` over initialized memory.
   ///
   /// The memory representing `count` instances starting at
-  /// `pointer` must be owned by the instance `owner`,
-  /// meaning that as long as `owner` is alive the memory will remain valid.
+  /// `pointer` must remain valid throughout the lifetime of
+  /// the newly-created `Span`.
   ///
   /// - Parameters:
   ///   - pointer: a pointer to the first initialized element.
   ///   - count: the number of initialized elements in the span.
   ///   - owner: a binding whose lifetime must exceed that of
   ///            the newly created `Span`.
+  @_disallowFeatureSuppression(NonescapableTypes)
   @_alwaysEmitIntoClient
+  @lifetime(immortal)
```

Comment:
Should be `@lifetime(start)`. The resulting Span depends on the validity of the pointer.